### ISSUE TITLE
Inconsistent behavoir during job creation between `1.1.0` and older version.

### ISSUE THREAD

**Issue by Andromelus** [saagie/api-saagie#124]
**Describe the bug**

My team has used an old, local, version of this library for some time. I am currently make updates to the program that uses the library and I want to upgrade saagieapi.

However, there is an inconsistent behavior between this new version (`1.1.0`) and the one we used to use. I do not remember the version we cloned. I **think** that it might be `0.7.4`, but unsure. You will find a zip of our local version attached to the issue.

### Behavior with old, local saagieapi

```python
from saagieapilocal.projects.saagie_api import SaagieApi

saagie = SaagieApi(url_saagie="MYURL",
                   id_platform=2,
                   user=my_user,
                   password=my_pwd,
                   realm="matmut"
                   )

catalog = "Matmut-PROD"
techno = "Python_MATMUT"
runtime = "3.5.2-0.0.3"
project_id = "PROJECT ID"
extra_tech = ""
extra_runtime = ""

info = saagie.create_job(
    job_name="job_name",
    project_id="1413945a-8bfc-4174-85f2-8b7e03bb70e1",
    file="version",
    description="",
    category="Extraction",
    technology_catalog=catalog,
    technology=techno,
    runtime_version=runtime,
    command_line="echo toto",
    release_note="release_note",
    extra_technology=extra_tech,
    extra_technology_version=extra_runtime
)

print(info)
```

which works as expected: The job is created with the technology of my custom catalog.

### Behavior with saagie api `1.1.0`

```python
from saagieapi import SaagieApi
import os

saagie = SaagieApi(url_saagie="URL",
                   id_platform=2,
                   user=login,
                   password=password,
                   realm="matmut"
                   )

catalog = "Matmut-PROD"
techno = "Python_MATMUT"
runtime = "3.5.2-0.0.3"
project_id = "PROJECT ID"
extra_tech = ""
extra_runtime = ""


saagie.jobs.create(
    job_name="job_name3",
    project_id="PROJECTID",
    file="version",
    description="",
    category="Extraction",
    technology_catalog=catalog,
    technology=techno,
    runtime_version=runtime,
    command_line="echo toto",
    release_note="release_note",
    extra_technology=extra_tech,
    extra_technology_version=extra_runtime
)
```

which throws:

```
Traceback (most recent call last):
  File "test.py", line 22, in <module>
    saagie.jobs.create(
  File "C:\Users\castelain.florianext\Documents\anaconda\lib\site-packages\saagieapi\jobs\jobs.py", line 289, in create    
    raise RuntimeError(
RuntimeError: ❌ Runtime 3.5.2-0.0.3 for technology Python_MATMUT does not exist in the catalog Matmut-PROD or is deprecated
```

### Catalog configuration

The technology, in my custom catalog, is not deprecated and does exist. Here is the catalog configuration:

```
contexts:
  - id: 3.5.2-0.0.3
    label: 3.5.2-1.0.1
    available: true
    recommended: true
    trustLevel: stable
    job:
      features:
        - type: COMMAND_LINE
          label: Command line
          mandatory: true
          comment: Linux shell command to launch the job.
          defaultValue: python {file} arg1 arg2
        - type: ARTIFACT
          label: Package
          mandatory: true
          comment: "Compatible upload file : .py or .zip (with optional requirements.txt for pip install)"
        - type: SCHEDULER
          label: Scheduled
          mandatory: true
```

**Expected behavior**

The creation of the job should be successful with the **newer** version of the library, as well as with older versions (as it is the case currently).

[saagieapilocal.zip](https://github.com/saagie/api-saagie/files/9028915/saagieapilocal.zip)

**Comment by saagie-service** [comment]
Trello card: [[#124] Inconsistent behavoir during job creation between `1.1.0` and older version.](https://trello.com/c/OZD0XzJt)

**Comment by Andromelus** [comment]
It **seems** that saagieapi is checking existance of the runtime by using the `label ` property defined in the catalog, instead of the `id` property.

**If** this is the case, it should be written somewhere. Moreover, the catalog definition should also have something called `runtime_name` or something related, to stay consistant with naming accross the entire perimeter.

**Comment by Andromelus** [comment]
In addition, there in an inconsistency between `upgrade` and `create` for `jobs.`

To check if a runtime is available, during `upgrade`, you do not check if the `available` key in the dictionnary exists:

https://github.com/saagie/api-saagie/blob/5df1cb63d44352f4fd11e4d13a2c39124dbc067e/saagieapi/jobs/jobs.py#L457

However, you check if it does exist during `create`:
https://github.com/saagie/api-saagie/blob/5df1cb63d44352f4fd11e4d13a2c39124dbc067e/saagieapi/jobs/jobs.py#L286

**Comment by tquiviger** [comment]
Issue has been fixed with release 1.1.2 and the wiki has been updated thanks to your feedbacks
